File: Assignment/Abstraction/FunctionNode.cs
using System;

namespace Assignment.Abstraction
{
    internal struct FunctionNode: INode
    {
        public FunctionNode(string function, INode argument)
        {
            this.Function = function;
            this.Argument = argument;
        }

        public string Function { get; }
        public INode Argument { get; }

        public string DisplayName()
        {
            return nameof(FunctionNode);
        }
    }
}

File: src/main/java/ce305/abstraction/expressions/FunctionNode.java
package ce305.abstraction.expressions;

import ce305.abstraction.INode;

public class FunctionNode implements INode {
    public final String function;
    public final INode argument;

    public FunctionNode(String function, INode argument) {
        super();
        this.function = function;
        this.argument = argument;
    }

    @Override
    public String getDisplayName() {
        return String.format("%(name: %)", FunctionNode.class.getName(), this.function);
    }
}